src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -75,9 +79,63 @@ class EventEntries extends React.Component<Props> {
     );
   }
 
-  recordIssueError(errorTypes: any[], errorMessages: string[]) {
+  async fetchDebugFile(query: string) {
+    const {api, organization, project} = this.props;
+    try {
+      const debugFiles = await api.requestPromise(
+        `/projects/${organization.slug}/${project.slug}/files/dsyms/`,
+        {
+          method: 'GET',
+          query: {
+            query,
+            file_formats: organization.features?.includes('android-mappings')
+              ? ['breakpad', 'macho', 'elf', 'pe', 'pdb', 'sourcebundle']
+              : undefined,
+          },
+        }
+      );
+      this.setState({hasProGuardError: debugFiles.length > 1});
+    } catch (error) {
+      Sentry.withScope(scope => {
+        scope.setLevel(Sentry.Severity.Error);
+        Sentry.captureException(error);
+        // do nothing, the UI will not display extra error details
+      });
+    }
+  }
+
+  checkProGuardError() {
+    const {event} = this.props;
+
+    // const threads =
+    //   event.entries.find(e => e.type === EntryType.THREADS)?.data?.values ?? [];
+    // const bestThread = findBestThread(threads);
+
+    const debugImages = event?.entries.find(e => e.type === EntryType.DEBUGMETA)?.data
+      .images as undefined | Array<Image>;
+
+    const proGuardImage = debugImages?.find(
+      debugImage => debugImage.type === 'proguard' && !!debugImage.uuid
+    );
+
+    if (!proGuardImage?.uuid) {
+      return;
+    }
+
+    this.fetchDebugFile(proGuardImage.uuid);
```

Comment:
This seem to cover 1 class of errors. When we have the `debug_meta` and the `uuid` correctly set in the event but the proguard file wasn't found.

The error message for this case could be:

`The mapping file for the specified proguard id XYZ was not found.`

This also means the plug did run, otherwise the proguard id wouldn't be in the event. But Maybe it failed to upload the mappings file. /cc @marandaneto might have insights here too.


We're not checking whether there are actually minified frames in the stack trace though. We could avoid this check altogether if there's no problem to report at all.

Reply:
tbh I'm not even sure if this is a real use case, because the uuid file is not even generated if the upload failed, unless you've made a dry run, but this is still valid because of that https://github.com/getsentry/sentry-cli/issues/914
after this change that we need, this use case will be valid, so I'd keep it

Reply:
@bruno-garcia proguard has many rules and one of them is, I can minify part of my App but not all, hence if you run a regex, you should do in the whole stack trace.

Reply:
you could minify class names but not packages, and many other granular configs.